test/python/compiler/test_disassembler.py

```diff
@@ -211,6 +224,163 @@ def test_multiple_conditionals_multiple_registers(self):
         self.assertEqual(circuits[0], qc)
         self.assertEqual({}, header)
 
+    def test_single_circuit_calibrations(self):
+        """Test that disassembler parses single circuit QOBJ calibrations (from QOBJ-level)."""
+        theta = Parameter('theta')
+        qc = QuantumCircuit(2)
+        qc.h(0)
+        qc.append(RXGate(np.pi), [0])
+        qc.append(RXGate(theta), [1])
+        qc = qc.assign_parameters({theta: np.pi})
+
+        with pulse.build() as h_sched:
+            pulse.play(pulse.library.Drag(1, 0.15, 4, 2), pulse.DriveChannel(0))
+
+        with pulse.build() as x180:
+            pulse.play(pulse.library.Gaussian(1, 0.2, 5), pulse.DriveChannel(1))
+
+        qc.add_calibration('h', [0], h_sched)
+        qc.add_calibration(RXGate(np.pi), [0], x180)
+        qc.add_calibration(RXGate(np.pi), [1], x180)
+
+        qobj = assemble(qc, FakeOpenPulse2Q())
+        dasm_circuits, _, _ = disassemble(qobj)
+
+        self.assertEqual(len(qc.calibrations), len(dasm_circuits[0].calibrations))
+        self.assertEqual(qc.calibrations.keys(), dasm_circuits[0].calibrations.keys())
+        self.assertEqual(all([
+            qc_cal.keys() == dasm_qc_cal.keys()
+            for qc_cal, dasm_qc_cal in
+            zip(qc.calibrations.values(), dasm_circuits[0].calibrations.values())]), True)
+        self.assertEqual(all([
+            _parametric_to_waveforms(qc_sched) == _parametric_to_waveforms(dasm_qc_sched)
+            for (_, qc_gate), (_, dasm_qc_gate) in
+            zip(qc.calibrations.items(), dasm_circuits[0].calibrations.items())
+            for qc_sched, dasm_qc_sched in zip(qc_gate.values(), dasm_qc_gate.values())]), True)
```

Comment:
This whole section is useful for multiple tests. Consider factoring this into a private method. This would have the extra benefit that in each test, I only have to understand the setup, and then there's something like `self.assertCircuitCalibrationsEqual(qc, output_circ)` which tells me explicitly what the test is intended to check. And the details of how to assert that two calibrations are the same is tucked away in the method.